Buddies/Utilities/AlgoliaSearch.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  AlgoliaSearch.swift
+//  Buddies
+//
+//  Created by Noah Allen on 2/12/19.
+//  Copyright © 2019 Jack and the Beans. All rights reserved.
+//
+
+/** Usage example:
+let search = AlgoliaSearch()
+let s = "2019/02/15".toDate()?.date
+let e = "2019/02/20".toDate()?.date
+search.searchActivities(startingAt: s, endingAt: e, atLocation: (41.1603988, -80.0866907)) { (activities: [String], err: Error?) in
+ 
+    if let error = err { print(error) }
+    print(activities)
+}
+**/
+
+import Foundation
+import InstantSearchClient
+
+protocol SearchIndex {
+    func search(_ query: Query, requestOptions: RequestOptions?, completionHandler: @escaping CompletionHandler) -> Operation
+}
+extension Index: SearchIndex {}
+
+protocol SearchClient {
+    func getIndex(withName indexName: String) -> SearchIndex
+}
+
+// Wrap the function so that we can use "SearchIndex" as a mock
+extension Client: SearchClient {
+    func getIndex(withName indexName: String) -> SearchIndex {
+        return index(withName: indexName)
+    }
+}
+
+class AlgoliaSearch {
+    static let ACTIVITY_INDEX = "BUD_ACTIVITIES"
+    var client: SearchClient
+
+    init (algoliaClient: SearchClient? = nil) {
+        self.client = algoliaClient != nil ? algoliaClient! : AlgoliaSearch.algoliaFactory()
+    }
+
+    // Constructs an instance of the Algolia Client SDK:
+    static func algoliaFactory() -> Client {
+        var algoliaAppID = "NOPE"
+        var algoliaApiKey = "NOPE"
```

Comment:
Is there any risk of error by just using "NOPE"?

Reply:
No - if the keys exist, it will use the keys. If the keys do not exist (which is basically impossible since it can't actually build the code without the keys file existing), it will use "NOPE" which would cause the client to not connect to a backend since four characters aren't long enough to be a key. I think that's what we want. Alternatively, I could use empty strings.

Reply:
Maybe empty strings? Or a more informative/specific string? Looks good as is though

Reply:
I've changed it to be empty strings, and also re-worked the factory test to assert that the key it initializes with is not an empty string.